source/vir/src/sst_to_air.rs

```diff
@@ -839,7 +900,7 @@ pub fn body_stm_to_air(
         &mut HashSet::new(),
         stm,
     );
-    let mut stmts = stm_to_stmts(ctx, &mut state, &stm);
+    let mut stmts = stm_to_stmts(ctx, &mut state, &stm, has_mut_params);
```

Comment:
Having a special "pre_snap: bool" parameter for has_mut_params seems awkward.  Can't you just insert StmtX::Snapshot(snapshot_ident(SNAPSHOT_PRE)) into stmts at position 0 here (after stm_to_stmts returns)?

Reply:
Indeed. Fixed.